fairlearn/metrics/_metric_frame.py

```diff
@@ -113,10 +171,12 @@ class MetricFrame:
         metric function name, with the values being the string-to-array-like dictionaries.
     """
 
+    @_deprecate_metric_frame_init
     def __init__(self,
-                 metric: Union[Callable, Dict[str, Callable]],
+                 *,
+                 metrics: Union[Callable, Dict[str, Callable]],
```

Comment:
Not sure I quite understand. The code in this PR issues a deprecation warning if you pass `metric` (it's handled by the decorator). You get the following message:
```pycon
>>> mf = MetricFrame(metric=skm.accuracy_score, y_true=y_true, y_pred=y_pred, sensitive_features=sf)
FutureWarning: The positional argument 'metric' has been replaced by a keyword argument 'metrics'. From version 0.8.0 passing it as a positional argument or as a keyword argument 'metric' will result in an error
```

Reply:
I see. I would rather not have this warning in the decorator. The decorator is nice to handle the positional argument deprecation, but we can easily deprecate an old parameter in the function itself, which would make the code more readable.

Reply:
I see. If I understand correctly, you'd like to have both `metrics` and `metric` as keyword arguments, but:
* in the numpy doc section use a [deprecation block](https://numpydoc.readthedocs.io/en/latest/format.html#deprecation-warning)
* in the constructor code, issue a warning

Reply:
If both `metrics` and `metric` are passed, raise `TypeError` or issue a warning and ignore `metric`?

Reply:
Yes, exactly

Reply:
Actually, there's an issue--if we want to have compatibility with the old API, we need to make `metrics` / `y_true` / `y_pred` formally optional, otherwise we'll run into this:
```pycon
>>> mf = MetricFrame(metric=skm.accuracy_score, y_true=y_true, y_pred=y_pred, sensitive_features=sf)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/mnt/c/Users/mdudik/projects/fairlearn/fairlearn/metrics/_metric_frame.py", line 71, in compatible_metric_frame_init
    **kwargs)
TypeError: __init__() missing 1 required keyword-only argument: 'metrics'
```
So my suggestion is to have the signature where `metrics` / `y_true` / `y_pred` are optional, but raise if any of them is `None`. Thoughts?

Reply:
@adrinjalali : when I tried to do this without a decorator, I found myself re-implementing standard Python error warnings around missing keyword-only arguments, so I've decided to go with the decorator approach, but to add `metric=None` in the signature and docs. (maybe that makes code actually more confusing?)